### ISSUE TITLE
Login fail & TypeError

### ISSUE THREAD

**Issue by iRonin** [dubocr/homebridge-tahoma#115]
On Jul 28/29 Tahoma stopped working for me and on 29th I checked it and I was getting `[Tahoma] Login fail: Bad credentials`. I was getting the same error on tahomalink.com, although Somfy Connect worked fine. I tried resetting password to no avail.

I noticed there was this plugin's update so I updated and restarted but then Homebridge would not start up because of this error:

```
[7/29/2019, 4:11:43 PM] TypeError: Cannot read property 'length' of null
    at Server.<anonymous> (/usr/local/lib/node_modules/homebridge/lib/server.js:407:44)
    at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/once.js:16:19
    at /homebridge/node_modules/homebridge-tahoma/index.js:93:6
    at TahomaPlatform.<anonymous> (/homebridge/node_modules/homebridge-tahoma/index.js:194:5)
    at /homebridge/node_modules/homebridge-tahoma/overkiz-api.js:189:4
    at Request._callback (/homebridge/node_modules/homebridge-tahoma/overkiz-api.js:262:25)
    at Request.self.callback (/homebridge/node_modules/request/request.js:186:22)
    at Request.emit (events.js:189:13)
    at Request.<anonymous> (/homebridge/node_modules/request/request.js:1163:10)
    at Request.emit (events.js:189:13)
    at IncomingMessage.<anonymous> (/homebridge/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:277:13)
    at IncomingMessage.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

I ended up disabling the plugin, so I lost all the configuration in HomeKit.

Today tahomalink.com worked fine, so I enabled the plugin and everything is back (plus there is an alarm, that I hadn't get before).

I'm not sure about this error above but the plugin should not spam Tahoma service on bad credentials, maybe the timeout should be much longer, and a notification (besides logs) would be awesome. But as always, loosing the devices as a result of disabling the plugin is the worst.

Anyway, it's great to have this plugin!

**Comment by dubocr** [comment]
Hi,

Sorry for this trouble.
I'll try to improve kind of error.
However, TaHoma might natively support Homekit until end of 2019 and all this trouble will be backward ;-).

Best

**Comment by iRonin** [comment]
> TaHoma might natively support Homekit until end of 2019 and all this trouble will be backward ;-).

Yeah, I read about it yesterday, but I'm sceptical and worried that they will mess things up, but maybe not. 

Your plugin has been a real help!

**Comment by dubocr** [comment]
It was initially planned for end of 2018 and they start to work on it begening of 2018 so with 2 years development and 1 year postpone we could expect they will be ready end of this year.

Wait and see ;-)

**Comment by dubocr** [comment]
Hi,

This is fixed in last release 0.3.11. Hope this will solve any future trouble. The plugin will normally never return empty accessory list anymore.
